api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDao.java

```diff
@@ -443,6 +453,49 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		return Optional.empty();
 	}
 	
+	/**
+	 * A handler for a {@link HasParam}
+	 *
+	 * @param propertyName the name of the property in the query to use
+	 * @param hasParam the {@link HasParam} to handle
+	 * @return a {@link Criterion} to be added to the query for the indicate has param
+	 */
+	protected Optional<Criterion> handleHas(String propertyName, HasParam hasParam) {
+		if (hasParam == null) {
+			return Optional.empty();
+		}
+		ArrayList<IQueryParameterType> orValues = new ArrayList<>();
+		String targetResourceType = null;
+		String paramReference = null;
+		String parameterName = null;
+		
+		String paramName = null;
+		targetResourceType = hasParam.getTargetResourceType();
+		
+		if (targetResourceType.getClass().equals(Observation.class)) {
```

Comment:
`targetResourceType` is a `String` so `targetResourceType.getClass()` will always return `java.lang.String`. What you actually want is:

```suggestion
		if (FhirConstants.OBSERVATION.equals(targetResourceType)) {
```